Generate and execute a separate test suite for each app module

haim-kermany opened the pull request:
## Description

1. run a config per module.
2. moving the instrumented classes to the output directory, makes sure that nothing is created at the cli directory
3. merging the path fix to the configs creation, and removing fix_config()

Related to # 170

## Type of Change

Please check the types of changes your PR introduces.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Refactoring (non-breaking code restructuring that preserves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build-related update (CI workflow, test cases)
- [ ] Documentation update
- [ ] Other (please describe):

## How Has This Been Tested?
a new test:

Please describe the tests that you ran to verify your changes. Provide instructions so that it can be replicated.
Please also list any relevant details for your test configuration.

test_generate_execute_multi_modules()
__assert_no_artifact_at_cli()

- [ ] Test A
- [ ] Test B

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

rachelt44 reviewed (changes requested):
1. `__assert_no_artifact_at_cli` should be a general method and not refer hardcoded to specific apps, but receive an app name as input (this would make more sense also if we want to include more apps in the future)

2. in `execute.py` line 38 you compute output_dir - but then in line 39 you computed it again instead of using the variable you computed on the previous line

3. Same comment for `generate.py` lines 45-46: use previously computed variable (and here replace order between the two lines)
4. `tkltest_unit.py` line 64: for consistency it is better that this line is printed in all cases, and if there are no modules then the app name is used. Also better to print "Generating/Executing tests for app/module {} using config file {}"  rather than the current message.

haim-kermany commented:
tnx for the comments:
1. fixed
4. fixed - please notice that at this stage we did not have the toml file, so I added is to the config. (made also the relevant fix in the unit tests)
2 +3. fixed - please notice that both get_output_dir() and cd_output_dir() needed the app_name and module_name as parameters.
so I just change that cd_output_dir() will also return the output_dir
